myapp/spec/system/tasks_spec.rb

```diff
@@ -79,61 +79,108 @@
-    before do
-      create(:task, name: 'Task1', priority: 'High', status: 'Done', duedate: '2024-02-10',created_at: Time.now- 3.days)
-      create(:task, name: 'Task2', priority: 'Low', status: 'In Progress' , duedate: '2024-02-23',created_at: Time.now- 2.days)
-      create(:task, name: 'Task3', priority: 'Medium', status: 'Not Started', duedate: '2024-02-12', created_at: Time.now- 1.days)
-    end
+  it 'invalid if one or more required contents are missing' do
+    task = Task.new(
+      name: '',
+      description: '',
+      priority: '',
+      status: '',
+      duedate: '2024-08-08'
+    )
+    expect(task).to be_invalid
+  end
 
-    it '3 tasks should be created' do
-      visit '/'
-      expect(Task.all.length).to eq 3
-    end
+  it 'valid if name length is 30' do
+    task = build(:task, name: 'a'*30)
+    expect(task).to be_valid
+  end
 
-    it 'by created_time desc/asc' do
-      visit '/'
-      click_link('created_at')
-      expect(page.body.index('Task3')).to be < page.body.index('Task2')
-      expect(page.body.index('Task2')).to be < page.body.index('Task1')
-      click_link('created_at')
-      expect(page.body.index('Task1')).to be < page.body.index('Task2')
-      expect(page.body.index('Task2')).to be < page.body.index('Task3')
-    end
+  it 'invalid if name is too long' do
+    task = build(:task, name: 'a'*31)
+    expect(task).to be_invalid
+  end
 
+  it 'valid if description length is 255' do
+    task = build(:task, description: 'a'*255)
+    expect(task).to be_valid
+  end
 
-    it 'by priority asc' do
-      visit '/'
-      click_link('Priority')
-      expect(page.body.index('Task1')).to be < page.body.index('Task3')
-      expect(page.body.index('Task3')).to be < page.body.index('Task2')
-      click_link('Priority')
-      expect(page.body.index('Task2')).to be < page.body.index('Task3')
-      expect(page.body.index('Task3')).to be < page.body.index('Task1')
-    end
+  it 'invalid if description is too long' do
+    task = build(:task, description: 'a'*256)
+    expect(task).to be_invalid
+  end
 
-    it 'by status asc' do
-      visit '/'
-      click_link('Status')
-      expect(page.body.index('Task1')).to be < page.body.index('Task2')
-      expect(page.body.index('Task2')).to be < page.body.index('Task3')
-      click_link('Status')
-      expect(page.body.index('Task3')).to be < page.body.index('Task2')
-      expect(page.body.index('Task2')).to be < page.body.index('Task1')
-    end 
-
-      it 'by duedate asc' do
-        visit '/'
-        click_link('Duedate')
-        expect(page.body.index('Task1')).to be < page.body.index('Task3')
-        expect(page.body.index('Task3')).to be < page.body.index('Task2')
-        click_link('Duedate')
-        expect(page.body.index('Task2')).to be < page.body.index('Task3')
-        expect(page.body.index('Task3')).to be < page.body.index('Task1')
-      end
+  it 'valid if task duedate is future' do
+    task = build(:task, duedate: '2024-08-08')
```

Comment:
This test will fail after August 8th, so how about like this?

```suggestion
    task = build(:task, duedate: Time.zone.tomorrow)
```

Reply:
Changed it. It works 😄